pymeasure/display/widgets/sequencer_widget.py

```diff
@@ -556,13 +557,14 @@ def load_sequence(self, *, filename=None):
         :param filename: Filename (string) of the to-be-loaded file.
         """
         append_flag = False
+
         if (filename is None) or (filename == ''):
             dialog = SequenceDialog()
-            dialog.exec()
-            append_flag = dialog.append_checkbox.checkState() == QtCore.Qt.CheckState.Checked
-            filenames = dialog.selectedFiles()
-            if filenames:
-                filename = filenames[0]
+            if dialog.exec():
+                append_flag = dialog.append_checkbox.checkState() == QtCore.Qt.CheckState.Checked
+                filenames = dialog.selectedFiles()
+                if filenames:
+                    filename = filenames[0]
```

Comment:
```suggestion
                filename = dialog.selectedFiles()[0]
```
I agree with your changes, but I think that when the dialog is accepted, a file must be selected, so I don't think the `if` make sense.

Reply:
I think, it's best to have just one file to avoid ambiguities.

Reply:
Yea I agree. Feel free to merge this, I added a fix for the tests a few minutes ago. I think the new_sequencer can be brought out of draft and to pymeasure. I'll close the sequencer PRs I made.